### PR TITLE
Change the ReadFromMessage method of the adaptive-step size integrators to be static

### PR DESCRIPTION
**Pull request by pleroy**
This is a follow-up of #2404.

### CODE REVIEW

File: integrators/embedded_explicit_generalized_runge_kutta_nyström_integrator_test.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@ namespace internal_embedded_explicit_generalized_runge_kutta_nyström_integrator
 using numerics::EstrinEvaluator;
 using numerics::LegendrePolynomial;
 using quantities::Abs;
+using quantities::Length;
```
Comment (eggrobin) [inline integrators/embedded_explicit_generalized_runge_kutta_nyström_integrator_test.cpp]: Why?

File: integrators/integrators_body.hpp [hunk 2]
```diff
@@ -8,141 +8,252 @@
...
-                            sprk_action)                                  \
-  PRINCIPIA_INTEGRATOR_CASE(YOSHIDA_1990_ORDER_8D,                        \
-                            吉田1990Order8D,                              \
-                            sprk_action)                                  \
-  PRINCIPIA_INTEGRATOR_CASE(YOSHIDA_1990_ORDER_8E,                        \
-                            吉田1990Order8E,                              \
+// All the case branches in a switch on the serialized adaptive-step size
+// integrator kind.  Depending on the nature of the integrator, each case branch
+// calls one of the given actions, which must be 1-argument macros.
+// It has not escape our notice that the acronym is a fair characterization of
```
Comment (eggrobin) [inline integrators/integrators_body.hpp]: Typo.

File: integrators/integrators.hpp [hunk 3]
```diff
@@ -211,21 +215,12 @@ class AdaptiveStepSizeIntegrator : public Integrator<ODE_> {
 
   virtual void WriteToMessage(
       not_null<serialization::AdaptiveStepSizeIntegrator*> message) const = 0;
+  template<typename S = typename ODE::SystemState,
+           typename = std::enable_if_t<base::is_serializable_v<S>>>
```
Comment (eggrobin) [inline integrators/integrators.hpp]: Why?

File: integrators/embedded_explicit_generalized_runge_kutta_nyström_integrator_test.cpp (hunk 1 above)
Comment (pleroy) [reply integrators/embedded_explicit_generalized_runge_kutta_nyström_integrator_test.cpp]: An aborted attempt at adding a test.

File: integrators/integrators.hpp (hunk 3 above)
Comment (pleroy) [reply integrators/integrators.hpp]: Changed to allow deserialization of integrators for non-serializable frames (both ASS and FSS).

**Pull request merged**